### PR TITLE
Timer vat

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
This is the swingSet side of the timerVat and wrapper. It now works reliably in scenario 2 through restarts.

### CODE REVIEW

**Comment by warner** [comment]
BTW feel free to land this as-is and create a new ticket to remind us to make those changes later

File: src/vats/vat-timerWrapper.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import harden from '@agoric/harden';
+import nat from '@agoric/nat';
```
[the remark this answers is not in the document]
Comment (Chris-Hibbert) [reply src/vats/vat-timerWrapper.js]: Done.

File: docs/timer.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+```
+await vats.timerWrapper~.registerTimerDevice(devices.timer);
+const timerService = vats.timerWrapper~.createTimerService();
+```
+
+Then users in the REPL can use the timerService to schedule wakeups.
+
+```
+const timestamp = timerService~.getCurrentTimestamp();
```
Comment (warner) [inline docs/timer.md]: do these examples need `await`? Or maybe the variable should be named `timestampP` or something to make it clear that this is a promise.

File: docs/timer.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+const handler = harden({wake(now) { console.log(`woke up ${now}`); }});
+const willWakeAt = timerService~.setWakeup(60, handler);
+```
+
+The handler will fire in somewhat after 60 seconds from now.
```
Comment (warner) [inline docs/timer.md]: grammar nit, maybe "will fire somewhat after" or "will fire in somewhat more than"

**Comment by warner** [comment]
looks good with those nits addressed, thanks for the docs

File: docs/timer.md (hunk 2 above)
Comment (warner) [reply docs/timer.md]: hm, I guess I'm imagining someone copying this code into their vat, in which case 'timestamp' isn't really a timestamp, it's a promise for one, and the difference should be clear in the docs (e.g. you couldn't do `timestamp + 30*60` to display the expected expiration time of a 30-minute contract). But we'll probably get more feedback from users, and can refine it later.

**Pull request merged**